sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -71,9 +73,17 @@ case class InMemoryRelation(
 
   override def computeStats(): Statistics = {
     if (batchStats.value == 0L) {
-      // Underlying columnar RDD hasn't been materialized, no useful statistics information
-      // available, return the default statistics.
-      Statistics(sizeInBytes = child.sqlContext.conf.defaultSizeInBytes)
+      children.filter(_.isInstanceOf[LogicalRelation]) match {
+        case Seq(c @ LogicalRelation(_, _, _, _), _) if c.conf.cboEnabled =>
```

Comment:
`InMemoryRelation` is a `logical.LeafNode`. I think it has no children?

Reply:
thanks, @viirya , yes, you're right! I misread the generated plan, working on it